packages/network_info_plus/network_info_plus/android/src/main/kotlin/dev/fluttercommunity/plus/network_info/NetworkInfo.kt

```diff
@@ -1,15 +1,42 @@
 package dev.fluttercommunity.plus.network_info
 
+import android.Manifest
+import android.app.Activity
+import android.content.pm.PackageManager
 import android.net.wifi.WifiInfo
 import android.net.wifi.WifiManager
+import android.os.Build
+import android.util.Log
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.PluginRegistry
 import java.net.*
 
+private const val permissionRequestCode = 100
+
 /** Reports network info such as wifi name and address. */
-internal class NetworkInfo(private val wifiManager: WifiManager?) {
+internal class NetworkInfo(private val wifiManager: WifiManager?) :
+    PluginRegistry.RequestPermissionsResultListener {
 
     private val wifiInfo: WifiInfo?
         get() = wifiManager?.connectionInfo
 
+    private var ongoingResult: MethodChannel.Result? = null
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ): Boolean {
+        if (requestCode == permissionRequestCode) {
+            val cachedResult = ongoingResult ?: return true
+            cachedResult.success(parseAuthStatus(grantResults).platformName)
+            ongoingResult = null
+        }
+        return true
+    }
+
```

Comment:
I personally would not include code to request the permissions in the plugin. There's plugins that can already do that better than what we can: e.g. https://pub.dev/packages/permission_handler